src/api/embed_helpers.cc

```diff
@@ -107,6 +112,17 @@ CommonEnvironmentSetup::CommonEnvironmentSetup(
   impl_->isolate = NewIsolate(impl_->allocator, &impl_->loop, platform);
   Isolate* isolate = impl_->isolate;
 
+  impl_->cpp_heap = CppHeap::Create(
+      platform,
+      CppHeapCreateParams {
+          {},
+          WrapperDescriptor(
+              BaseObject::kEmbedderType,
+              BaseObject::kSlot,
+              0x90df)
```

Comment:
I don't think the exact value matters a whole lot.  In Electron, Chromium will initialize the heap with `0x1`.  Node addons making use of `cppgc` don't need to specify the value, it's automatically set by `cppgc::MakeGarbageCollected`.

What does matter, is that it doesn't match the value used in the embedder fields for non-cppgc objects.  That was why https://github.com/nodejs/node/commit/8b0e5b19bd0 updated the value in the first place.

We could certainly expose a method that uses this fixed value that embedders can use.

Reply:
Question is: where?  It could potentially be something extra done in `SetIsolateUpForNode` (that _is_ called by Electron, so I guess it would need to be optionally done based on something in `IsolateSettings`).  Or it could be a new method.

Reply:
> that _is_ called by Electron, so I guess it would need to be optionally done based on something in `IsolateSettings`

I think that would seem okay to me :+1: `IsolateSettings` was added specifically so that Electron can customize this a bit more iirc.

Reply:
One issue with this is that the `CppHeap` needs to be owned by someone.  With the changes I have thus far, it's either owned by `CommonEnvironmentSetup::Impl`, or `NodeMainInstance`.

`SetIsolateUpForNode` doesn't have any place to own the `CppHeap`.

Reply:
I guess we could make it part of the `MultiIsolatePlatform`’s job to attach/detach the cpp heap in `RegisterIsolate`/`UnregisterIsolate`? And if we need this behavior to be toggle-able in the default implementation, we could add an extra argument to `MultiIsolatePlatform::Create()`…

Reply:
So far it doesn't seem necessary to toggle this in `NodePlatform`. Also AFAICT, where Electron would need us to skip the cppgc initialization it should also configure `ProcessInitializationFlags::kNoInitializeNodeV8Platform` too? So I am thinking if we can just use that flag to decide whether we need to do cppgc initialization, as in the `NodePlatform` we also have a reference to the page allocator that can be used to initialize cppgc, and this way we can put the cppgc-related registration/initialization cold all in the NodePlatform implementation if we also attach/detach the cpp heap in `RegisterIsolate/UnregisterIsolate`.

Reply:
On a closer look, `IsolateData` might be a better place to own the CppHeap. And it should be safe to attach/detach in its constructor/destructor.

Reply:
Apparently our own tests don't actually treat `IsolateData` as per-isolate even though they are meant to be that way...I've sent https://github.com/nodejs/node/pull/48450 to address that. If that doesn't work, we could still go with RegisterIsolate/UnregisterIsolate.

Reply:
Moved the CppHeap to IsolateData, which now detects if the cpp heap is already created before creating a cpp heap.